### PR TITLE
[Feature] Add better email error

### LINKED ISSUE #9733: 🐛 Email validation gap on support page
## 🐛 Bug

It is possible to enter an email address that will pass frontend validation but fail backend validation.  Unfortunately, the error message is very vague when this happens.

## 🦋 Expected Behaviour

Option 1 is just improve the error message so that the user knows what went wrong and can try to correct it.  We can't control Laravel's or different browser's email address validation.

Option 2 is that we could consider removing all validation on the email field.  It could be argued that the support form is too important to allow to fail or that users may want to intentionally obfuscate their email address in a way that a human might be able to interpret.

## 📋 Steps to Reproduce

The example I found in the server logs was `validemail*temp*@gmail.com` in Chrome, as shown in the screenshot.

## 📸 Screenshot

image

## ✅ Acceptance Criteria

- [ ] Improve backend validation error toast
- [ ] Respond with a 400 error response, not 500, if that's what we get from api

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #9733

## 👋 Introduction

This branch adds a better error to the support page when an "invalid email" code is being returned by Freshdesk.  I ran the suggested error message past Marc and got a "looks good".

## 🧪 Testing

1. Set up your environment for Freshdesk (either set up a junk account or copy the values from prod).

- [ ] An invalid email address triggers the new error message.
- [ ] Any other error (like bad API key) triggers the existing default error message.
- [ ] A successful ticket submission triggers the existing success toast.

## 📸 Screenshot

image

### CODE REVIEW

File: apps/web/src/lang/fr.json [hunk 1]
```diff
@@ -2667,6 +2667,10 @@
     "defaultMessage": "Non démontré suffisamment dans le cadre de la présente méthode d’évaluation – faire passer à de plus amples mises à l’essai. (Ne s’applique qu’aux occasions où les points sont cumulatifs à l’échelle des méthodes d’évaluation. Ne s’applique pas dans les cas où les candidats doivent répondre à certains critères essentiels à chaque étape).",
     "description": "Note for when an assessment was unsuccessful but put on hold"
   },
+  "DOn3Hm": {
+    "defaultMessage": "Adresse courriel non valide. Essayez de nouveau ou envoyez un courriel, o <anchorTag>{emailAddress}</anchorTag>.",
```
Comment (mnigh) [inline apps/web/src/lang/fr.json:2671]: Based on the English equivalent, I think this is a typo.

```suggestion
    "defaultMessage": "Adresse courriel non valide. Essayez de nouveau ou envoyez un courriel à <anchorTag>{emailAddress}</anchorTag>.",
```

Comment (petertgiles) [reply apps/web/src/lang/fr.json:2671]: Thanks!  Fix checks out on deepl.

File: apps/web/src/pages/SupportPage/components/SupportForm/SupportForm.tsx [hunk 2]
```diff
@@ -309,13 +276,52 @@ const SupportFormApi = () => {
...
+    const errorCode = `${response.status} - ${response.statusText}`;
+    logger.error(`Failed to submit ticket: ${JSON.stringify(responseBody)}`);
+
+    // default error message if we don't recognize the error
+    let errorMessage = defineMessage({
+      defaultMessage:
+        "Sorry, something went wrong. Please email <anchorTag>{emailAddress}</anchorTag> and mention this error code: {errorCode}.",
+      id: "rNVDaA",
+      description: "Support form toast message error",
+    });
```
Comment (yonikid15) [inline apps/web/src/pages/SupportPage/components/SupportForm/SupportForm.tsx:300]: I think we're moving to defining messages like this outside the component (just above usually) so they aren't getting redeclared.

**Review by yonikid15 — APPROVED** [review]
Nice work looks good!

Invalid email:
image

Bad api key:
image

File: apps/web/src/pages/SupportPage/components/SupportForm/SupportForm.tsx (hunk 2 above)
Comment (petertgiles) [reply apps/web/src/pages/SupportPage/components/SupportForm/SupportForm.tsx]: Good idea, thanks!
[pull messages out of component](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9864/commits/d136f73078ddec9942ca324e46b6a7afa55ed501)

**Pull request merged**